frontend/packages/core/src/AppProvider/short-link.tsx

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import { useParams } from "react-router-dom";
+import type { clutch as IClutch } from "@clutch-sh/api";
+
+import { useNavigate } from "../navigation";
+import { client } from "../Network";
+import type { ClutchError } from "../Network/errors";
+
+interface ShortLinkProps {
+  hydrate: (data: IClutch.shortlink.v1.IShareableState[]) => void;
+  onError: (error: string) => void;
+  setLoading: (loading: boolean) => void;
+}
+
+const fetchData = async (hash, hydrate, onError) => {
+  const requestData: IClutch.shortlink.v1.IGetRequest = { hash };
+
+  return client
+    .post("/v1/shortlink/get", requestData)
+    .then(response => {
+      const { path, state } = response.data as IClutch.shortlink.v1.IGetResponse;
+
+      hydrate(state);
+      return path;
+    })
+    .catch((error: ClutchError) => {
+      // eslint-disable-next-line no-console
+      console.error(error?.status?.text);
```

Comment:
if we pass back the error object we can probably remove this

Reply:
Can do